src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -565,6 +578,34 @@ def resolve_imports_and_packages(
             max_batch_size,
             source_code_display=source_code_display,
         )
+
+        if pickled_by_reference and not isinstance(
+            func, (BuiltinMethodType, BuiltinFunctionType)
+        ):
+            try:
+                source_code = inspect.getsource(func)
+            except Exception as exc:
+                raise SnowparkClientExceptionMessages.SOURCE_CODE_EXTRACTION_ERROR(
+                    func, exc
+                )
+            with tempfile.TemporaryDirectory() as tempdir:
+                abs_path = os.path.join(
+                    tempdir, f"{func.__module__.replace('.', '/')}.py"
```

Comment:
hmm the file separator on windows is `\`. Does it matter? Though I don't see any test failing on windows.

Reply:
I mean, should we use this https://docs.python.org/3/library/os.html#os.sep